### DISCUSSION TITLE
RTL not working?

### DISCUSSION THREAD

**Discussion by ashikns** [microsoft/fluentui#28436] [Q&A] [answered]
I wrapped a `HorizontalBarChart` in a ThemeProvider and set `dir='rtl'` on it.

```
<ThemeProvider dir='rtl'>
    <HorizontalBarChart data={data}
      variant={HorizontalBarChartVariant.AbsoluteScale} />
</ThemeProvider>
```

However it doesn't seem to have any effect? When I step through `HorizontalChart.base.tsx`, the call to `getRTL()` does not receive the theme (`theme` is undefined inside the function), so it always returns false for rtl.

I'm almost certain I'm using ThemeProvider wrong, but from the docs it looks like this should just work? Please tell me what I'm missing here.

**Comment by khmakoto** [comment]
@ashikns you should provide `rtl` as part of the theme and not as the html `dir` attribute:

```ts
<ThemeProvider theme={{ rtl: true }}>
    <HorizontalBarChart data={data}
      variant={HorizontalBarChartVariant.AbsoluteScale} />
</ThemeProvider>
```

Let me know if that makes it work for you.

**Reply by ashikns** [reply]
@khmakoto Thank you for the quick response. I tried this and it seems to have did _something._

Without rtl:
![image](https://github.com/microsoft/fluentui/assets/6162428/4361e039-50c5-45fe-913f-7bc29a23aead)

With `theme={{ rtl: true }}`:
![image](https://github.com/microsoft/fluentui/assets/6162428/ec21e7a9-9670-4710-ab74-0030d88ae6f9)

So it looks like `chartTitle` has been affected, but not the bars or `barLabels`. Specifically I'm setting the breakpoint in HorizontalChart.base.tsx and it still seems like `_isRTL` is false inside it. At least from the code it looks like if it is true `barLabel` should appear on the opposite side as well.
![image](https://github.com/microsoft/fluentui/assets/6162428/49fe92f1-7ba1-48e8-8d6d-3ce8dbe7b36c)


To provide some context I'm trying to horizontally flip the bar chart. I can set `styles={{ chartWrapper: { transform: 'scaleX(-1)' } }}` and that works (with or without rtl):
![image](https://github.com/microsoft/fluentui/assets/6162428/b98a85f3-a91e-4b2a-b12c-43b557003617)

but as you can see it flips the text rendering as well. If there's any other way to do this that'll work for me as well.

**Reply by khmakoto** [reply]
Seems like RTL support is not properly implemented yet in those charts.

CCing @AtishayMsft who might be able to provide more information here.

**Reply by ashikns** [reply]
I manually copied over `HorizontalChart.base.tsx` (and the bare minimum files to make it work, so excuse the broken titles and such, but it demonstrates the point) and manually flipped the x position. That works as I expect it to, the `barLabel`s are exactly flipped:
![image](https://github.com/microsoft/fluentui/assets/6162428/4b7f21d7-bb92-48ac-bb1f-8223337c08cc)

Sorry, one more edit, in case this helps accelerate the bug fix. If I pass `this.props.theme` to the `getRTL` call in [line 43](https://github.com/microsoft/fluentui/blob/34af1fe15779b6069c9081d5ad60cbb0289746b3/packages/react-charting/src/components/HorizontalBarChart/HorizontalBarChart.base.tsx#L43) on `HorizontalBarChart.base.tsx`, it works beautifully as I expect:
![image](https://github.com/microsoft/fluentui/assets/6162428/a5d6f9f8-9a84-4bb6-a778-57eaa677e21e)

**Reply by AtishayMsft** [reply, marked as answer]
@ashikns, There was a change in the past to allow rtl to be set via themes https://github.com/microsoft/fluentui/commit/dc551ba339019ca89bde77ae327770072c673758. 

Charting package did not adopt this change. We will add it to our backlog. Thanks for letting us know.

Meanwhile you can set rtl using below methods that charting honors.
1. Set `dir='rtl'` to root element or body element of your html page.
2. Set the rtl state of the page to true using `@fluentui/react/lib/Utilities setRTL` function.
3. Set `isRTL=1` in your browser local storage.

**Reply by ashikns** [reply]
Unfortunately that doesn't work for me. I need to set rtl on a per component basis.

**Reply by AtishayMsft** [reply]
In that case you are welcome to fix the issue as you suggested above in charting code and raise a PR. We will be happy to guide and merge your fix.
